pkg/network/usm/shared_libraries.go

```diff
@@ -282,7 +291,16 @@ func (w *soWatcher) Start() {
 
 				for _, r := range w.rules {
 					if r.re.Match(path) {
-						w.registry.register(root, libPath, lib.Pid, r)
+						if quircksWaitLib(root + libPath) {
```

Comment:
trying to understand, is `libnetty_tcnative_linux_` is part of the openssl/gnutls regexes? can you please add an example of the path?

Reply:
now seeing you added `libnetty_tcnative_linux_.*.so` rule
it makes me wonder, why don't you implement a wrapper to `addHooks` instead adding all of this specific log here?
I think it will be cleaner to review and read, and efficiency-wise, we'll not compare strings for every match

Reply:
it's embedded in the netty-tcnative jar libnetty_tcnative_linux_x86_64.so but the will copied to the tempdir with a unique name like `/tmp/libnetty_tcnative_linux_x86_64_<u64 / probably epoch ns>.so`